src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -538,7 +554,57 @@ export class RoslynLanguageServer {
             childProcess = cp.spawn(serverPath, args, cpOptions);
         }
 
-        return childProcess;
+        // Timeout promise used to time out the connection process if it takes too long.
+        const timeout = new Promise<undefined>((resolve) => {
+            RAL().timer.setTimeout(resolve, 10000);
```

Comment:
If we're using the timeout for both the wait and connect, we might want to give a bit bigger of a buffer (this is 10seconds currently right?)

Maybe we start with 30s?